### PR TITLE
We can use tls

### PR DESCRIPTION
**Pull request by tdesvenain**

### CODE REVIEW

File: docs/changes.txt [hunk 1]
```diff
@@ -78,17 +87,17 @@ Thanks to Christian Zagrodnick for these changes.
 
 3.3.1 (12 Dec 2008)
 -------------------
-  
+
```
Comment (cjw296) [inline docs/changes.txt]: These whitespace changes are really annoying. Why are they in this diff? If you must include them, please separate them out into their own commit.

File: mailinglogger/MailingLogger.py [hunk 2]
```diff
@@ -39,7 +40,8 @@ def __init__(self,
                  headers=None,
                  template=None,
                  charset='utf-8',
-                 content_type='text/plain'):
+                 content_type='text/plain',
+                 secure=None):
```
Comment (cjw296) [inline mailinglogger/MailingLogger.py]: Why `secure` here rather than `use_tls`?

File: mailinglogger/datatypes.py [hunk 3]
```diff
@@ -23,25 +23,27 @@ def headers(self):
...
-                             self.section.send_empty_entries,
-                             self.section.flood_level,
-                             self.section.username,
-                             self.section.password,
-                             self.section.ignore,
-                             self.headers(),
-                             self.section.template,
-                             self.section.charset,
-                             self.section.content_type)
+                             mailhost=self.mailhost(),
```
Comment (cjw296) [inline mailinglogger/datatypes.py]: What value are these keywords adding?

**Comment by cjw296** [comment]
Hmm, there is actually a .travis.yml on this, any thoughts on why this branch wasn't tested?

**Comment by tdesvenain** [comment]
Hi @cjw296, maybe it's because i developed it in a fork ? I can't create a branch in simplistix repo, can i ?

**Comment by cjw296** [comment]
pull requests are normally built, see #11 .

File: mailinglogger/datatypes.py (hunk 3 above)
Comment (tdesvenain) [inline mailinglogger/datatypes.py]: It's about consistency with MailingLoggerHandlerFactory.create_loghandler method, a for code auto-documentation

File: mailinglogger/MailingLogger.py (hunk 2 above)
Comment (tdesvenain) [inline mailinglogger/MailingLogger.py]: because this is not the signature of use_tls. use_tls is a boolean, secure is a tuple
i called it secure because it is called secure in SMTPHandler (parent class)

File: docs/changes.txt (hunk 1 above)
Comment (tdesvenain) [inline docs/changes.txt]: Ok i'll split 1015bdc3a81d3f1c7997a8f466b9924cdbb2db9d into into two separate commits

**Comment by tdesvenain** [comment]
Okay, maybe it's because i made th pr on 3.8 branch. I may try to pr it on master and then backport it on 3.8, may i ?

**Comment by cjw296** [comment]
sure, master makes sense...

**Comment by tdesvenain** [comment]
looks obvious, indeed, but the problem is: zope config support (component.xml) has been removed in master, and we need it, so we work with 3.8.
so btw, what do you think about forward-port component.xml in master ?

**Comment by cjw296** [comment]
Wow, Zope, really? You using Plone per chance?

**Comment by tdesvenain** [comment]
You guessed it :)

**Comment by cjw296** [comment]
I'm afraid you're in a really bad place. The 2.8 branch is dead, and I don't want to add back all the zconfig stuff on master. What might be a way forward is if you do a zope-mailinglogger or plone.mailinglogger package that wraps the master branch and adds back the ZConfig stuff, and then do the changes to master here to get your TLS support before relesaing your plone.mailinglogger package.

That said, why do you want TLS? Seems like a very odd configuration. Have a local MTA, log to that, and have it speak TLS or whatever when relaying on to an external MTA.

**Comment by cjw296** [comment]
nb: edits to the above,  trying to make it clear what I'm suggesting and failing...

**Comment by cjw296** [comment]
@tdesvenain - where did you got with this in the end? I just released 4.0.0 today, so interested in your thoughts. Closing this one for now, though, as it's very out of date.
